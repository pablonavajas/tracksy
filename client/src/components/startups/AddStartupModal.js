import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addStartup } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const AddStartupModal = ({ addStartup }) => {
  const [name, setName] = useState('');
  const [website, setWebsite] = useState('');
  const [ownership, setOwnership] = useState('');
  const [board, setBoard] = useState('');
  const [investment_1, setInvestment_1] = useState('');
  const [type_1, setType_1] = useState('');
  const [date_1, setDate_1] = useState('');
  const [investment_2, setInvestment_2] = useState('');
  const [type_2, setType_2] = useState('');
  const [date_2, setDate_2] = useState('');

  const onSubmit = () => {
    if (
      name === '' ||
      website === '' ||
      ownership === '' ||
      board === '' ||
      investment_1 === '' ||
      type_1 === '' ||
      date_1 === '' ||
      investment_2 === '' ||
      type_2 === '' ||
      date_2 === ''
    ) {
      M.toast({
        html: 'Please enter NA in the fields that are not applicable'
      });
    } else {
      const newStartup = {
        name,
        website,
        ownership,
        board,
        investment_1,
        investment_2,
        type_1,
        type_2,
        date_1,
        date_2
      };

      addStartup(newStartup);

      // Clear fields
      setName('');
      setWebsite('');
      setOwnership('');
      setBoard('');
      setInvestment_1('');
      setDate_1('');
      setType_1('');
      setInvestment_2('');
      setDate_2('');
      setType_2('');
    }
    //console.log('Submit button Clicked');
  };

  return (
    <div id="add-startup-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter New Startup</h4>
        {/** Name */}
        <div className="row">
          <div className="input-field">
            <input
              id="name"
              type="text"
              name="name"
              value={name}
              onChange={e => setName(e.target.value)}
            />
            <label>Startup Name</label>
          </div>
        </div>

        {/** Website */}
        <div className="row">
          <div className="input-field">
            <input
              id="website"
              type="text"
              name="name"
              value={website}
              onChange={e => setWebsite(e.target.value)}
            />
            <label>Website</label>
          </div>
        </div>

        {/** Ownership */}
        <div className="row">
          <div className="input-field">
            <input
              id="ownership"
              name="ownership"
              type="text"
              value={ownership}
              onChange={e => setOwnership(e.target.value)}
            />
            <label>Ownership (%)</label>
          </div>
        </div>

        {/** Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_1"
              name="investment_1"
              type="text"
              value={investment_1}
              onChange={e => setInvestment_1(e.target.value)}
            />
            <label>Investment 1</label>
          </div>
        </div>
        {/** Type of Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_1"
              name="type_1"
              type="text"
              value={type_1}
              onChange={e => setType_1(e.target.value)}
            />
            <label>Type of Investment 1</label>
          </div>
        </div>
        {/** Closing Date Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_1"
              name="date_1"
              value={date_1}
              type="date"
              onChange={e => setDate_1(e.target.value)}
            />
            <label>Close Date</label>
          </div>
        </div>

        {/** Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_2"
              name="investment_2"
              type="text"
              value={investment_2}
              onChange={e => setInvestment_2(e.target.value)}
            />
            <label>Investment 2</label>
          </div>
        </div>

        {/** Type of Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_2"
              name="type_2"
              type="text"
              value={type_2}
              onChange={e => setType_2(e.target.value)}
            />
            <label>Type of Investment 2</label>
          </div>
        </div>

        {/** Closing Date Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_2"
              name="date_2"
              value={date_2}
              type="date"
              onChange={e => setDate_2(e.target.value)}
            />
            <label>Close</label>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="input-field col s12">
          <select
            id="board"
            name="board"
            value={board}
            className="browser-default"
            onChange={e => setBoard(e.target.value)}
          >
            <option value="" disabled>
              Select Board
            </option>
            <option value="Member">Member</option>
            <option value=" - "> - </option>
            <option value="Observer"> Observer </option>
          </select>
        </div>
      </div>
      <div className="modal-footer">
        <a
          href="#!"
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: '60%',
  height: '75%'
};

AddStartupModal.propTypes = {
  addStartup: PropTypes.func.isRequired
};

export default connect(null, { addStartup })(AddStartupModal);
