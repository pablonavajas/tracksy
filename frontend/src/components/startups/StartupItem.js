import React from "react";
import PropTypes from "prop-types";
import CurrencyFormat from "react-currency-format";
var moment = require("moment");

import { connect } from "react-redux";
import { deleteStartup, setCurrent } from "../../actions/startupsActions";
import { ClickIcon } from "../smallComponents/displayComponents";
import { Chip } from "../smallComponents/displayComponents";

export const StartupItem = ({ startup, setCurrent, deleteStartup }) => {
  return (
    <tr>
      <th className="center">
        <a href={`https://${startup.website}`}>
          <img
            src={`http://logo.clearbit.com/${startup.website}`}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">
        <Chip data={startup.ownership} />
      </td>
      <td className="center">
        <Chip data={startup.board} />
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip
              data={
                <CurrencyFormat
                  value={investment.value}
                  displayType={"text"}
                  thousandSeparator={true}
                  prefix={investment.currency}
                />
              }
            />
          </div>
        ))}
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={investment.investmentType} />
          </div>
        ))}
      </td>
      <td>
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={moment(investment.date).format("DD/MM/YYYY")} />
          </div>
        ))}
      </td>
      <td>
        <ClickIcon
          href="#startup-overview"
          setFunction={setCurrent}
          val={startup}
          iconName="dehaze"
        />
      </td>
      <td>
        <ClickIcon
          href="#edit-startup-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="edit"
        />
      </td>
      <td>
        <ClickIcon
          href="#add-investments-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="attach_money"
        />
      </td>
      {/* Display grey button with no functionality if some financials have already been added */}
      {startup.financials.length === 0 ? (
        <td>
          <ClickIcon
            href="#add-kpi-names-modal"
            setFunction={setCurrent}
            val={startup}
            iconName="insert_chart"
          />
        </td>
      ) : (
        <td>
          <a
            id="unclickableIcon"
            href="!#"
            className="secondary-content"
            onClick={(e) => e.preventDefault()}
          >
            <i className="material-icons grey-text">insert_chart</i>
          </a>
        </td>
      )}
      <td>
        <a
          id="deleteIcon"
          href="#!"
          onClick={(e) => {
            e.preventDefault();
            deleteStartup(startup.id);
          }}
          className="secondary-content"
        >
          <i className="material-icons blue-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

StartupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

// Inject App Level state, so that
// component updates upon an update of info
// (e.g. added investment)
const mapStateToProps = (state) => ({
  startups: state.startup.startups,
});

export default connect(mapStateToProps, { setCurrent, deleteStartup })(
  StartupItem
);
