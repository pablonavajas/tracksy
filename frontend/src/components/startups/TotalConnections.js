import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {getConnections} from  '../../actions/connectionsActions';


const TotalConnections = ({ current, connections, getConnections }) => {
    const [name, setName] = useState("");
    const [website, setWebsite] = useState("");

    useEffect(() => {
    getConnections();
  }, [connections, current]);

  if (connections === null) {
    return (
      <div className="progress">
        <div className="indeterminate"></div>
      </div>
    );
  }

  return (
  // Section: Stats
  
  <section className="section section-stats center">
    <nav>
    <div className="nav-wrapper blue-grey darken-4">
      <ul id="nav-mobile" className="left hide-on-med-and-down">
        <li><a href="#startup-overview">Overview</a></li>
        <li><a href="#hiring">Hiring</a></li>
        <li className="active"><a href="#total-connections">Connections</a></li>
      </ul>
    </div>
  </nav>
  
  <div className="col s12 m6 l4">
          <div className="card">
            <div className="card-content">
              <table className="striped">
                <thead>
                  <tr className = "center">
                    <th>Name</th>
                    <th>Description</th>
                    <th>URL</th>
                  </tr>
                </thead>
                <tbody>
                   {connections.map((c) => (
                     <tr key={c.id}>
                       <td> {c.name} </td>
                       <td> {c.description} </td>
                       <td> {c.url} </td>
                   </tr>
                   ))}
                </tbody>
              </table>
            </div>
        </div>
        </div>

      </section>
  );
};


TotalConnections.propTypes = {
  current: PropTypes.object,
  getConnections:PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  connections: state.connections,
  current: state.startup.current
});

export default connect(mapStateToProps, { getConnections })(TotalConnections);
