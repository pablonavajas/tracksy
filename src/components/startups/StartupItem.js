import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { deleteStartup, setCurrent } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const startupItem = ({ startup, setCurrent, deleteStartup }) => {
  const onDelete = () => {
    deleteStartup(startup.id);
    M.toast({ html: 'Startup Deleted' });
  };

  return (
    <tr>
      <th className="center">
        <a href={'https://' + startup.website}>
          <img
            src={'http://logo.clearbit.com/' + startup.website}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">{startup.ownership}</td>
      <td className="center">{startup.board}</td>
      <td>{startup.investment_1}</td>
      <td className="center">{startup.type_1}</td>
      <td>{startup.date_1}</td>
      <td className="center">{startup.investment_2}</td>
      <td className="center">{startup.type_2}</td>
      <td>{startup.date_2}</td>
      <td>
        <a
          href="#edit-startup-modal"
          onClick={() => setCurrent(startup)}
          className="secondary-content modal-trigger"
        >
          <i className="material-icons grey-text">edit</i>
        </a>
      </td>
      <td width="50">
        <a href="#!" onClick={onDelete} className="secondary-content">
          <i className="material-icons grey-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

startupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired
};

export default connect(null, { setCurrent, deleteStartup })(startupItem);
