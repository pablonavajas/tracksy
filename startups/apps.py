from django.apps import AppConfig


class TracksyappConfig(AppConfig):
    name = 'startups'
