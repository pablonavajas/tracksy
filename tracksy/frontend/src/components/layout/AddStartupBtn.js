import React from "react";

//Button that triggers AddStartupModal
const AddStartupBtn = () => {
  return (
    <div className="fixed-action-btn">
      <a
        href="#add-startup-modal"
        className="btn-floating  waves-effect waves-light btn-large red darken-1 modal-trigger"
      >
        <i className="large material-icons">add</i>
      </a>
    </div>
  );
};

export default AddStartupBtn;
