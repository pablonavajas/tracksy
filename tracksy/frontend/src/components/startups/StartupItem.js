import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { deleteStartup, setCurrent } from "../../actions/startupsActions";
import CurrencyFormat from "react-currency-format";
var moment = require("moment");

const StartupItem = ({ startup, setCurrent, deleteStartup }) => {
  return (
    <tr>
      <th className="center">
        <a href={"https://" + startup.website}>
          <img
            src={"http://logo.clearbit.com/" + startup.website}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">
        <div className="chip center">{startup.ownership}</div>
      </td>
      <td className="center">
        <div className="chip center">{startup.board}</div>
      </td>

      <td className="center">
        {startup.investments.map((investment, i) => (
          <div className="center" key={investment.id}>
            <div className="chip center">
              <p>
                {i + 1}
                {". "}
                <CurrencyFormat
                  value={investment.value}
                  displayType={"text"}
                  thousandSeparator={true}
                  prefix={investment.currency}
                />
              </p>
            </div>
          </div>
        ))}
      </td>
      <td className="center">
        {startup.investments.map(investment => (
          <div className="center" key={investment.id}>
            <div className="chip center">{investment.investmentType}</div>
          </div>
        ))}
      </td>
      <td>
        {startup.investments.map(investment => (
          <div className="center" key={investment.id}>
            <div className="chip center">
              {moment(investment.date).format("DD/MM/YYYY")}
            </div>
          </div>
        ))}
      </td>
      <td>
        <a
          href="#add-investments-modal"
          onClick={() => setCurrent(startup)}
          className="secondary-content modal-trigger"
        >
          <i className="material-icons grey-text">attach_money</i>
        </a>
      </td>
      <td>
        <a
          href="#edit-startup-modal"
          onClick={() => setCurrent(startup)}
          className="secondary-content modal-trigger"
        >
          <i className="material-icons grey-text">edit</i>
        </a>
      </td>
      <td>
        <a
          href="#!"
          onClick={() => deleteStartup(startup.id)}
          className="secondary-content"
        >
          <i className="material-icons grey-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

StartupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired
};

export default connect(null, { setCurrent, deleteStartup })(StartupItem);
