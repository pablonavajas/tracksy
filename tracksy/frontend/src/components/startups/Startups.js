import React, { useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import StartupItem from "./StartupItem";
import { getStartups } from "../../actions/startupsActions";
import AddStartupBtn from "./../layout/AddStartupBtn";

export const Startups = ({ startup: { startups, loading }, getStartups }) => {
  //everything inside useEffect hook, gets called at page start up if [] is empty
  useEffect(() => {
    getStartups();
    // eslint-disable-next-line
  }, []);

  if (loading || startups === null) {
    return <h4>Loading ...</h4>;
  }

  if (!loading && startups.length === 0) {
    return (
      <div>
        <h5 className="left">No logs to show...</h5>
        <AddStartupBtn />
      </div>
    );
  }

  return (
    <div>
      <div className="row">
        <div className="col s12">
          <div className="card">
            <div className="card-content">
              <span className="card-title center">Portfolio Companies</span>
              <table className="striped centered">
                <thead>
                  <tr>
                    <th>Logo</th>
                    <th>Ownership (%)</th>
                    <th>Board</th>
                    <th className="center">Investment 1</th>
                    <td className="center">Type</td>
                    <td className="center">Closed</td>
                    <th className="center">Investment 2</th>
                    <td className="center">Type</td>
                    <td className="center">Closed</td>
                    <th></th>
                    <th></th>
                  </tr>
                </thead>
                <tbody>
                  {startups.map(startup => (
                    <StartupItem startup={startup} key={startup.id} />
                  ))}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <div
        className="right"
        style={{
          margin: "-60px 50px"
        }}
      >
        <AddStartupBtn />
      </div>
    </div>
  );
};

Startups.propTypes = {
  startup: PropTypes.object.isRequired,
  getStartups: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  startup: state.startup //state.startups is from rootReducer from index.js
});

//first argument (to get anything from AppLevelState and get it into the component)
export default connect(mapStateToProps, { getStartups })(Startups);
