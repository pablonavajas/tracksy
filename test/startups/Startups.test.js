import React from "react";
import { Provider } from "react-redux";

import { mount, shallow } from "enzyme";
import "../setupTests";
import toJson from "enzyme-to-json";
import { Startups } from "../../tracksy/frontend/src/components/startups/Startups";
import StartupItem from "../../tracksy/frontend/src/components/startups/StartupItem";
import ConnectedStartups from "../../tracksy/frontend/src/components/startups/Startups";
import Preloader from "../../tracksy/frontend/src/components/layout/Preloader";
import AddStartupBtn from "../../tracksy/frontend/src/components/layout/AddStartupBtn";

import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
const mockStore = configureMockStore([thunk]);

//Snapshot Test
describe("Startups component ", () => {
  let wrapper;
  const startups = [
    { id: 0, name: "company1" },
    { id: 1, name: "company2" },
  ];

  const loading = false;
  const startup = { startups, loading };

  const mockFn = jest.fn();

  it("renders correctly", () => {
    wrapper = shallow(<Startups startup={startup} getStartups={mockFn} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders two StartupItem components as expected", () => {
    wrapper = shallow(
      <Startups startup={{ startups, loading }} getStartups={mockFn} />
    );
    expect(wrapper.find(StartupItem)).toHaveLength(2);
  });

  it("shows preloader when loading prop is true", () => {
    wrapper = shallow(
      <Startups startup={{ startups, loading: true }} getStartups={mockFn} />
    );
    expect(wrapper.contains(<Preloader />)).toBe(true);
  });

  it("shows preloader when startups prop is null", () => {
    wrapper = shallow(
      <Startups startup={{ startups: null, loading }} getStartups={mockFn} />
    );
    expect(wrapper.contains(<Preloader />)).toBe(true);
  });

  it("says no startups to show (and displays the add button), when starups prop lenght is 0", () => {
    wrapper = shallow(
      <Startups startup={{ startups: [], loading }} getStartups={mockFn} />
    );
    expect(
      wrapper.contains(<h5 className="left">No startups to show...</h5>)
    ).toBe(true);

    expect(wrapper.contains(<AddStartupBtn />)).toBe(true);
  });
});

describe("Startups component ", () => {
  const mockFn = jest.fn();
  const startups = [
    { id: 0, name: "company1", investments: [], financials: [] },
    { id: 1, name: "company2", investments: [], financials: [] },
  ];
  const loading = false;
  const startup = { startups, loading };
  const initialState = {
    startup,
  };

  const store = mockStore(initialState);
  let wrapper;

  it("correctly imports startup and loading from the store through mapStateToProps", () => {
    wrapper = shallow(<ConnectedStartups getStartups={mockFn} store={store} />);
    expect(wrapper.props().children.props.startup).toEqual(startup);
  });

  it("calls getStartup upon initialisation", () => {
    //mount instead of shallow, because shallow doesn't call React Hooks
    wrapper = mount(
      <Provider store={store}>
        <Startups getStartups={mockFn} startup={startup} />
      </Provider>
    );
    expect(mockFn).toHaveBeenCalled();
  });
});
