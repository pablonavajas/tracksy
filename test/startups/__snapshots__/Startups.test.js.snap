// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Startups component  renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s12"
    >
      <div
        className="card"
      >
        <div
          className="card-content"
        >
          <span
            className="card-title center"
          >
            Portfolio Companies
          </span>
          <table
            className="striped centered"
          >
            <thead>
              <tr>
                <th>
                  Name
                </th>
                <th>
                  Ownership (%)
                </th>
                <th>
                  Board
                </th>
                <th
                  className="center"
                >
                  Investment
                </th>
                <th
                  className="center"
                >
                  Type
                </th>
                <th
                  className="center"
                >
                  Closed
                </th>
                <th />
                <th />
                <th />
                <th />
                <th />
              </tr>
            </thead>
            <tbody>
              <Connect(StartupItem)
                key="0"
                startup={
                  Object {
                    "id": 0,
                    "name": "company1",
                  }
                }
              />
              <Connect(StartupItem)
                key="1"
                startup={
                  Object {
                    "id": 1,
                    "name": "company2",
                  }
                }
              />
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="right"
    style={
      Object {
        "margin": "-60px 50px",
      }
    }
  >
    <AddStartupBtn />
  </div>
</div>
`;
